src/regressor.jl

```diff
@@ -12,23 +18,30 @@ build(model::NeuralNetworkRegressor, rng, shape) =
 fitresult(model::NeuralNetworkRegressor, chain, y) = (chain, nothing)
 
 function MLJModelInterface.predict(model::NeuralNetworkRegressor,
-                                   fitresult,
-                                   Xnew)
-    chain  = fitresult[1]
+    fitresult,
+    Xnew)
+    chain = fitresult[1]
     Xnew_ = reformat(Xnew)
-    return [chain(values.(tomat(Xnew_[:,i])))[1]
-                for i in 1:size(Xnew_, 2)]
+    return [chain(values.(tomat(Xnew_[:, i])))[1]
+            for i in 1:size(Xnew_, 2)]
 end
 
 MLJModelInterface.metadata_model(NeuralNetworkRegressor,
-               input=Table(Continuous),
-               target=AbstractVector{<:Continuous},
-               path="MLJFlux.NeuralNetworkRegressor")
+    input=Union{AbstractMatrix{Continuous},Table(Continuous)},
+    target=AbstractVector{<:Continuous},
+    path="MLJFlux.NeuralNetworkRegressor")
 
 
 # # MULTITARGET NEURAL NETWORK REGRESSOR
 
+"""
+    shape(model::MultitargetNeuralNetworkRegressor, X, y)
+
+A private method that returns the shape of the input and output of the model for given data `X` and `y`.
+"""
 function shape(model::MultitargetNeuralNetworkRegressor, X, y)
+    X = X isa Matrix ? Tables.table(X) : X
+    y = y isa Matrix ? Tables.table(y) : y
     n_input = Tables.schema(X).names |> length
     n_output = Tables.schema(y).names |> length
     return (n_input, n_output)
```

Comment:
Sorry for not catching this earlier, but wrapping a matrix as a table seems a convoluted  way to get num  columns. 

What about something like this?
```suggestion
ncols(X::AbstractMatrix) = size(X, 2)
ncols(X) = Tables.columns(X)) |> Tables.columnnames |> length
shape(model::MultitargetNeuralNetworkRegressor, X, y) = (ncols(X), ncols(y))
```

(Note, I've also dumped `Tables.schema` in favour of different Tables methods, which should be more reliable: `schema` should only be applied to `Tables.rows(X)` or `Tables.columns(X)` and can give `nothing` if not implemented for a particular table. Unfortunately, one cannot avoid `Tables.rows(X)` or `Tables.columns(X)` to get reliably get the number of columns through Tables.jl API, as far as I can understand.)

Reply:
That makes more sense, yes!